src/Schema/Storage/AbstractTypeRegistry.php

```diff
@@ -21,7 +23,24 @@ abstract class AbstractTypeRegistry
      */
     public static function get(string $typename)
     {
-        return static::fromCache($typename);
+        try {
+            return static::fromCache($typename);
+        } catch (Exception $e) {
+            // Try to rebuild the whole schema as fallback.
```

Comment:
We need to make sure we respect `SilverStripe\GraphQL\Controller::autobuildEnabled()`
```suggestion
            if (!Controller::has_curr() || !(Controller::curr() instanceof GraphQLController) || !Controller::curr()->autobuildEnabled()) {
                throw $e;
            }
            // Try to rebuild the whole schema as fallback.
```

The class names here assume the new use statements:
```php
use SilverStripe\Control\Controller;
use SilverStripe\GraphQL\Controller as GraphQLController;
```

Note that the `Controller::has_curr()` here is probably redundant - but please include it. I'd rather be safe than sorry here.

Reply:
Updated